test/test_generator.rb

```diff
@@ -0,0 +1,47 @@
+require "find"
+require "yaml"
+
+require "minitest/autorun"
+
+require 'util/generator_helpers'
+
+class TestGenerator < Minitest::Test
+  include GeneratorTestHelpers
+
+  def setup
+    @default_options = [
+      'centos6-64mdca',
+    ]
+  end
+
+  def test_default_options
+    hash = run_cli_with_options(@default_options)
+    hosts = hash['HOSTS']
+
+    assert_includes(hosts, "centos6-64-1")
+    centos6 = hosts['centos6-64-1']
+    ["master", "database", "dashboard", "agent"].each do |role|
+      assert_includes(centos6['roles'], role)
+    end
+  end
+
+  def test_pe_dir_option
+    skip "pe_dir option not yet tested"
+  end
+
+  def test_fixtures
+    Find.find 'test/fixtures' do |f|
+      if File.directory?(f)
+        next
+      end
+
+      fixture_file = File.open(f, "r")
+      fixture_hash = YAML.load(fixture_file)
+
+      options = fixture_hash["options_string"]
+      options = options.split
+      test_hash = run_cli_with_options(options)
+      assert_equal(fixture_hash["expected_hash"], test_hash)
```

Comment:
This is probably a good place to document (for instance via the 3rd argument to `assert_equal`) what this failure means. I believe it means "the behavior of the generator has changed from the snapshotted behavior".

If that's true folks are also going to want guidance on how/when to re-snapshot things so that they can take the changed behavior as a new baseline.